src/main.rs

```diff
@@ -0,0 +1,188 @@
+use std::env;
+use std::fmt;
+use std::fs::{self,File};
+use std::io::{self,Read};
+use std::os::unix::fs as unix_fs;
+use std::path::{Path,PathBuf};
+use std::process::Command;
+
+struct PathError {
```

Comment:
```suggestion
#[derive(Debug)]
struct PathError {
```

I tend to always derive debug - it's a good habit to get into